enterprise/views.py

```diff
@@ -110,6 +110,44 @@
 ENTERPRISE_GENERAL_ERROR_PAGE = 'enterprise/enterprise_error_page_with_messages.html'
 
 
+CATALOG_API_CONFIG_ERROR_CODE = 'ENTGDS004'
+CUSTOMER_DOES_NOT_EXIST_ERROR_CODE = 'ENTGDS008'
+CONTENT_ID_DOES_NOT_EXIST_ERROR_CODE = 'ENTGDS000'
+LICENSED_ENROLLMENT_ERROR_CODE = 'ENTGDS100'
+NO_CONSENT_RECORD_ERROR_CODE = 'ENTGDS0002'
+REDIRECT_URLS_MISSING_ERROR_CODE = 'ENTGDS003'
+
+DSC_ERROR_MESSAGES_BY_CODE = {
+    CATALOG_API_CONFIG_ERROR_CODE: 'Course catalog api configuration error.',
+    CUSTOMER_DOES_NOT_EXIST_ERROR_CODE: 'No record found for Enterprise customer.',
+    CONTENT_ID_DOES_NOT_EXIST_ERROR_CODE: 'The course or program with the given id does not exist.',
+    LICENSED_ENROLLMENT_ERROR_CODE: 'Failed to enroll licensed learner into course.',
+    NO_CONSENT_RECORD_ERROR_CODE: 'There is no consent record, or consent is not required.',
+    REDIRECT_URLS_MISSING_ERROR_CODE: 'Required request values missing for action to be carried out.',
+}
+
+
+def _log_error_message(error_code, exception=None, info_only=False, **kwargs):
+    """
+    Helper to log standardized error messages about DSC API failures.
+    """
+    error_message = DSC_ERROR_MESSAGES_BY_CODE[error_code]
+    log_message = (
+        '[Enterprise DSC API] {error_message} '
+        'error_code: {error_code}, '
+        'exception: {exception}'
+    ).format(error_message=error_message, error_code=error_code, exception=exception)
+
+    for key, value in kwargs.items():
+        log_message += ', {key}: {value}'.format(key=key, value=value)
+
+    if info_only:
+        LOGGER.info(log_message)
+    else:
+        LOGGER.error(log_message)
```

Comment:
Should this be LOGGER.exception if the exception is present so we capture the stack trace?

Reply:
We're spitting out a lot of info in these log messages, I don't think a `logger.exception()` adds much.  Also, you can only get into these code paths by making a request directly to the GET or POST handler - there's nothing interesting higher up the stack, just Django request-handling stuff.